logstash-core/lib/logstash/output_delegator.rb

```diff
@@ -138,9 +149,13 @@ def worker_multi_receive(events)
   def do_close
     @logger.debug("closing output delegator", :klass => self)
 
-    @worker_count.times do
-      worker = @worker_queue.pop
-      worker.do_close
+    if @threadsafe
+      @workers.each(&:do_close)
+    else
+      worker_count.times do
```

Comment:
This is not possible because the pipeline does not start closing outputs or filters until all worker threads are stopped. https://github.com/elastic/logstash/blob/master/logstash-core/lib/logstash/pipeline.rb#L442